client/my-sites/plugins/marketplace/marketplace-domain-upsell/index.tsx

```diff
@@ -0,0 +1,171 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import DomainPicker from '@automattic/domain-picker';
+import { useShoppingCart } from '@automattic/shopping-cart';
+import CalypsoShoppingCartProvider from 'calypso/my-sites/checkout/calypso-shopping-cart-provider';
+import { domainRegistration } from 'calypso/lib/cart-values/cart-items';
+import { DomainSuggestions } from '@automattic/data-stores';
+import {
+	MARKETPLACE_FLOW_ID,
+	ANALYTICS_UI_LOCATON_MARKETPLACE_DOMAIN_SELECTION,
+} from 'calypso/my-sites/plugins/marketplace/constants';
+import { Button } from '@wordpress/components';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+
+import './styles.scss';
+import { getProductsList, isProductsListFetching } from 'calypso/state/products-list/selectors';
+
+type DomainSuggestion = DomainSuggestions.DomainSuggestion;
+import { fillInSingleCartItemAttributes } from 'calypso/lib/cart-values';
+
+function Header() {
+	return (
+		<div className="marketplace-domain-upsell__header domains__header">
+			<h1 className="marketplace-domain-upsell__header marketplace-title">Choose a domain</h1>
+			<h2 className="marketplace-domain-upsell__header marketplace-subtitle">
+				Yoast SEO requires a top level domain to index your site and help you rank higher.
+			</h2>
+		</div>
+	);
+}
+
+function DomainPickerContainer( { onDomainSelect, selectedDomain } ) {
+	return (
+		<DomainPicker
+			header={ <Header /> }
+			analyticsUiAlgo={ ANALYTICS_UI_LOCATON_MARKETPLACE_DOMAIN_SELECTION }
+			analyticsFlowId={ MARKETPLACE_FLOW_ID }
+			onDomainSelect={ onDomainSelect }
+			currentDomain={ selectedDomain }
+			hideRecommendationLabel={ true }
+		/>
+	);
+}
+
+function MarketplaceShoppingCart( { onAddDomainToCart, selectedDomain } ) {
+	const { responseCart, isLoading } = useShoppingCart();
+	const { products, sub_total_display } = responseCart;
+
+	return isLoading ? (
+		<div></div>
+	) : (
+		<>
+			<h1 className="marketplace-domain-upsell__shopping-cart marketplace-title">Your cart</h1>
+			<div className="marketplace-domain-upsell__shopping-cart basket-items">
+				{ products.map( ( product ) => {
+					return (
+						<div
+							key={ product.product_name }
+							className="marketplace-domain-upsell__shopping-cart basket-item"
+						>
+							<div>{ product.product_name }</div>
+							<div>{ product.item_subtotal_display }</div>
+						</div>
+					);
+				} ) }
+			</div>
+			<hr />
```

Comment:
Could we achieve this by using styles or dedicated components?

Reply:
Can you elaborate on this @tyxla, do we have any dedicated customizable components for <hr /> ?
Also do you mean to add a new styled component here for hr? Is there any additional advantage to add a styled component here? Really liked to know. Are we switching to a convention of using styled components across calypso?

Reply:
I think we're talking about different things here.

What I meant was that we could use a different component instead of a raw `<hr />` here. For example, we could put the contents into two `<CompactCard />` components and that will help split the contents semantically, too - see the last example [here](https://wpcalypso.wordpress.com/devdocs/design/card). 

Or maybe `<HrWithText />`, `<ListEnd />`, or even `HorizontalRule` from [Gutenberg](https://github.com/WordPress/gutenberg/blob/086b77ed409a70a6c6a6e74dee704851eff812f2/packages/primitives/src/horizontal-rule/README.md).

Essentially, the idea is to reuse as much as possible from the existing components, rather than introducing our own versions. Using the same design language is essential in the long run, and is very valuable when we update that design language and we want the updates to propagate to the entire app.

Reply:
Thanks for the feedback, I have tried to rely as much as possible on existing components, however I did resize and play around to make it more suited to this page.